nexus/src/app/crucible.rs

```diff
@@ -262,6 +264,193 @@ impl super::Nexus {
+            .maybe_get_crucible_region(log, dataset, region_id)
+            .await?
+        {
+            Some(region) => {
+                info!(
+                    log,
+                    "confirmed the region exists";
+                    "dataset" => ?dataset.id(),
+                    "region" => ?region,
+                );
+
+                region
+            }
+
+            None => {
+                error!(
+                    log,
+                    "region does not exist!";
+                    "dataset" => ?dataset.id(),
+                    "region" => ?region_id,
+                );
+
+                return Err(Error::invalid_request(format!(
+                    "dataset {:?} region {:?} does not exist!",
+                    dataset.id(),
+                    region_id,
+                )));
+            }
+        };
+
+        info!(
+            log,
+            "contacting crucible agent to confirm snapshot exists";
+            "dataset" => ?dataset.id(),
+            "region" => ?region_id,
+            "snapshot" => ?snapshot_id,
+        );
+
+        let snapshot = match self
+            .maybe_get_crucible_snapshot(log, dataset, region_id, snapshot_id)
+            .await?
+        {
+            Some(snapshot) => {
+                info!(
+                    log,
+                    "confirmed the snapshot exists";
+                    "dataset" => ?dataset.id(),
+                    "region" => ?region.id,
+                    "snapshot" => ?snapshot,
+                );
+
+                snapshot
+            }
+
+            None => {
+                // snapshot does not exist!
+                error!(
+                    log,
+                    "snapshot does not exist!";
+                    "dataset" => ?dataset.id(),
+                    "region" => ?region_id,
+                    "snapshot" => ?snapshot_id,
+                );
+
+                return Err(Error::invalid_request(format!(
+                    "dataset {:?} region {:?} snapshot {:?} does not exist!",
+                    dataset.id(),
+                    region_id,
+                    snapshot_id,
+                )));
+            }
+        };
+
+        let client = self.crucible_agent_client_for_dataset(dataset)?;
+        let dataset_id = dataset.id();
+
+        // Request the running snapshot start, polling until the state
+        // transitions from Requested to Created
+
+        let create_running_snapshot = || async {
+            let running_snapshot = match ProgenitorOperationRetry::new(
+                || async {
+                    client
+                        .region_run_snapshot(
+                            &RegionId(region_id.to_string()),
+                            &snapshot_id.to_string(),
+                        )
+                        .await
+                },
+                || async { self.crucible_agent_gone_check(dataset_id).await },
+            )
```

Comment:
does `cargo fmt` let this live on the same line as the `let`?

Reply:
As the `let` on line 366 you mean?